### ISSUE TITLE
Type checking needed in "$contains" filter query

### ISSUE THREAD

**Issue by kamelat** [nuxt/content#2158]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.16.0
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  yarn@1.22.19
- Builder:          vite
- User Config:      devtools, modules, content
- Runtime Modules:  @nuxt/content@2.7.0, @nuxtjs/tailwindcss@6.8.0
- Build Modules:    -
------------------------------

### Reproduction

In contents/authors/username.yml :

```
firstname: Firstname
lastname: Lastname
username: username
organization: Organization
role: Role
description: description
``` 
In content/posts/post.md using Front Matter:
```
---
title: Title
description: Description
authors: ["username", "username2"]
tags: ["tag1","tag2","tag3"]
categories: ['cat1',"cat2","cat3"]
---
# Title

content
```
In app.vue:
```
<template>
  <div>
    <NuxtLayout>
        <NuxtPage/>
    </NuxtLayout>
  </div>
</template>
```

In page/authors/[:username].vue : 
```
 <template>
    <div>{{ author.firstname + "  " + author.lastname }}</div>
    <div v-for="post in posts">
        {{ post.title }}
    </div>
</template>

<script setup>
const { path } = useRoute()
const author = await queryContent('authors').where({ _path: path }).findOne()
const posts = await queryContent('posts').where({ 'authors': { $contains: ""}}).find()
</script>
```

Start the server and go to the page http://localhost:3000/authors/username

### Describe the bug

The "contains" operator does not handle correctly null, undefined and empty string values. It always matches and fetches all the contents ("posts" in this case).

**Comment by Barbapapazes** [comment]
Hello,

What must be the behavior when contains is null?

**Comment by kamelat** [comment]
In my opinion, it would make sense for a given key with a $contains operator to process the query as follows:

- an empty string value must fetch the contents whose the given key contains an empty string value
- the null and undefined values should return "undefined" or raise a TypeError error

What do you think @Barbapapazes  ?

**Comment by Barbapapazes** [comment]
Here the reproduction. In fact, the empty string has a strange behavior.

**Comment by Barbapapazes** [comment]
The issue is that empty array with `every` will always return true.

![image](https://github.com/nuxt/content/assets/45267552/207510ee-5ca0-4a46-b350-d57b7e86d3a0)

**Comment by kamelat** [comment]
Indeed, it comes from "includes" method which always returns true with an empty string:

<a href="https://github.com/nuxt/content/blob/main/src/runtime/query/match/index.ts">
<img width="468" alt="image" src="https://github.com/nuxt/content/assets/26351444/fb805f1e-f934-4c90-8089-c552768f6b66">
</a>
